### DISCUSSION TITLE
Run github action step unconditionally

### DISCUSSION THREAD

**Discussion by jmoguillansky-gpsw** [community/community#27220] [Actions]
<p>Hi,<br>
I have a github action with a job which contains the following steps:</p>
<ol>
<li>build the code</li>
<li>upload the build log.</li>
</ol>
<p>The problem is that if step 1 fails, then step 2 doesn’t run, so I can’t see the build log!</p>
<p>What’s the correct approach?</p>
<p>Thanks,<br>
Jeff</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Add an <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#always"><code>always()</code> condition</a> to step 2:</p>
<pre><code class="lang-yaml">- name: upload build log
  if: ${{ always() }}
  # ...
</code></pre>
